src/gui/editors/AutomationEditor.cpp

```diff
@@ -862,6 +925,56 @@ void AutomationEditor::mouseMoveEvent(QMouseEvent * mouseEvent )
 				}
 			break;
 			}
+			case EditMode::EditTangents:
+			{
+				// If we moved the mouse past the beginning correct the position in ticks
+				posTicks = std::max(posTicks, 0);
+
+				if (m_mouseDownLeft)
+				{
+					if (m_action == Action::MoveTangent)
+					{
+						timeMap& tm = m_clip->getTimeMap();
+						auto it = tm.find(m_draggedTangentTick);
+
+						// Safety check
+						if (it != tm.end())
+						{
+							// Calculate new tangent
+							float y = m_draggedOutTangent
+								? yCoordOfLevel(OUTVAL(it))
+								: yCoordOfLevel(INVAL(it));
+							float dy = m_draggedOutTangent
+								? y - mouseEvent->y()
+								: mouseEvent->y() - y;
+							float dx = std::abs(posTicks - POS(it));
+							float newTangent = dy / std::max(dx, 1.0f);
+
+							if (m_draggedOutTangent)
+							{
+								it.value().setOutTangent(newTangent);
+							}
+							else
+							{
+								it.value().setInTangent(newTangent);
+							}
+						}
+					}
+				}
+				else if (m_mouseDownRight)
+				{
+					if (m_action == Action::ResetTangents)
+					{
```

Comment:
You could just say:
```suggestion
				else if (m_mouseDownRight && m_action == Action::ResetTangents)
				{
```

Reply:
I thought separating the mouse conditional from the action one would make it more readable and easier to change in the future if new actions are added/changed, though it does make it more verbose and nested. I changed to the single conditional on this last commit and can keep it if it's preferred.